### PR TITLE
fix: CircleCI config proper parallelism/bucketing

### PR DESCRIPTION
**Pull request by edulop91**
We were not properly doing parallelism/workflows. CircleCI config generated could be DRYer, we can make improvements later.

### CODE REVIEW

File: templates/circleci/.circleci/config.yml.tmpl [hunk 1]
```diff
@@ -2,30 +2,29 @@
 # Make improvements in fogg, so that everyone can benefit.
 {{- $circleCI := . }}
 version: 2
-
 working_directory: &working_directory ~/fogg
 
 jobs:
-  build:
+{{- range $idx, $testBucket := $circleCI.TestBuckets }}
+  terraform_{{$idx}}:
```
Comment (jgadling) [inline templates/circleci/.circleci/config.yml.tmpl]: Curious why circle's built-in parallelism won't work here?

Comment (edulop91) [reply templates/circleci/.circleci/config.yml.tmpl]: I guess they have many forms of built=in parallelism. Are you referring to the command line globbing/splitting?

Comment (jgadling) [reply templates/circleci/.circleci/config.yml.tmpl]: Essentially, yes. The splitting by timings in particular looks pretty useful.

Comment (edulop91) [reply templates/circleci/.circleci/config.yml.tmpl]: It would work. We can make that improvement later?

**Pull request merged**